### PR TITLE
feat(sqla-core): Add support for rendering Database Specific queries 

### PR DESCRIPTION
**Pull request by cybojenix**
*Issue:* #290

*Description of changes:*

This makes the sqla-core plugin aware of how to compile a database specific query.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

fixes: #290

### CODE REVIEW

**Comment by cybojenix** [comment]
I'll restructure this PR after the [sqla 1.4 PR](https://github.com/aws/aws-xray-sdk-python/pull/289) is merged

**Comment by srprash** [comment]
@cybojenix Thanks. I'll take a look once it is ready. :)

**Comment by cybojenix** [comment]
Over to you, @srprash

File: tests/ext/sqlalchemy_core/test_postgres.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    return 'postgresql://{user}@{host}:{port}/{db}'.format(
+        user=dsn['user'],
+        host=dsn['host'],
+        port=dsn['port'],
+        db=dsn['database'],
+    )
+
+
+@pytest.fixture()
+def engine(postgres_db):
```
Comment (srprash) [inline tests/ext/sqlalchemy_core/test_postgres.py:32]: Would it be possible to re-use the `engine` specified in the `test_base`? Not a deal-breaker, just a good-to-have.

**Review by srprash — COMMENTED** [review]
Just a minor comment. Otherwise LGTM! Thanks!

**Comment by srprash** [comment]
And could you also update the branch? Thanks!

**Comment by cybojenix** [comment]
Both addressed

**Comment by srprash** [comment]
Awesome! thanks

**Review by srprash — APPROVED** [review]

**Pull request merged**